rlp/main.cpp

```diff
@@ -215,61 +185,117 @@ int main(int argc, char** argv)
+			("base-16", "Treat input RLP as hex encoded data.")
+			("keccak,k", "Output Keccak-256 hash only.")
+			("base-64", "Treat input RLP as base-64 encoded data.")
+			("64", "Treat input RLP as base-64 encoded data.")
+			("bin,b", "Treat input RLP as raw binary data.")
+			("base-256", "Treat input RLP as raw binary data.")
+			("quiet,q", "Don't place additional information on stderr.")
+			("help,h", "Print this help message and exit.")
+			("version,V", "Show the version and exit.");
+	po::options_description allowedOptions("Allowed options");
+	allowedOptions.add(generalOptions).add(modesOptions);
+	po::parsed_options parsed = po::command_line_parser(argc, argv).options(allowedOptions).allow_unregistered().run();
+	vector<string> unrecognisedOptions = collect_unrecognized(parsed.options, po::include_positional);
+	po::variables_map vm;
+	po::store(parsed, vm);
+	po::notify(vm);
+	for (size_t i = 0; i < unrecognisedOptions.size(); ++i)
 	{
-		string arg = argv[i];
-		if (arg == "-h" || arg == "--help")
-			help();
-		else if (arg == "render")
+		string arg =  unrecognisedOptions[i];
+		if (arg == "render")
 			mode = Mode::Render;
 		else if (arg == "create")
 			mode = Mode::Create;
-		else if ((arg == "-i" || arg == "--indent") && i + 1 < argc)
-			prefs.indent = argv[++i];
-		else if (arg == "--hex-ints")
-			prefs.hexInts = true;
-		else if (arg == "--string-ints")
-			prefs.stringInts = true;
-		else if (arg == "--ascii-strings")
-			prefs.forceString = prefs.forceHex = false;
-		else if (arg == "--force-string")
-			prefs.forceString = true;
-		else if (arg == "--force-hex")
-			prefs.forceHex = true, prefs.forceString = false;
-		else if (arg == "--force-escape")
-			prefs.escapeAll = true;
-		else if (arg == "-n" || arg == "--nice")
-			prefs.forceString = true, prefs.stringInts = false, prefs.forceHex = false, prefs.indent = "  ";
 		else if (arg == "list")
 			mode = Mode::ListArchive;
 		else if (arg == "extract")
 			mode = Mode::ExtractArchive;
 		else if (arg == "assemble")
 			mode = Mode::AssembleArchive;
-		else if (arg == "-L" || arg == "--lenience")
-			lenience = true;
-		else if (arg == "-D" || arg == "--dapp")
-			encrypt = true, encoding = Encoding::Base64;
-		else if (arg == "-V" || arg == "--version")
-			version();
-		else if (arg == "-q" || arg == "--quiet")
-			quiet = true;
-		else if (arg == "-x" || arg == "--hex" || arg == "--base-16")
-			encoding = Encoding::Hex;
-		else if (arg == "-k" || arg == "--keccak")
-			encoding = Encoding::Keccak;
-		else if (arg == "--64" || arg == "--base-64")
-			encoding = Encoding::Base64;
-		else if (arg == "-b" || arg == "--bin" || arg == "--base-256")
-			encoding = Encoding::Binary;
-		else if (arg == "-e" || arg == "--encrypt")
-			encrypt = true;
 		else if (inputFile.empty())
 			inputFile = arg;
 		else
 			otherInputs.push_back(arg);
 	}
+	if (vm.count("help")) {
+		cout << "Usage rlp <mode> [OPTIONS]" << endl << modesOptions
```

Comment:
I'm not expert here, but shouldn't the program_options be able to output that for us?

Reply:
fixed